src/jep/Jep.java

```diff
@@ -129,7 +132,7 @@ public void run() {
                      * while another thread is in python, then the thread state
                      * can get messed up leading to stability/GIL issues.
                      */
```

Comment:
I think I fixed the issue with the zombie thread. I tested it with `Thread.sleep` calls in a couple different locations and it's all executing in the desired order, with the interpreter thread stopping nicely.

For what it's worth, a colleague mentioned that using a `java.util.concurrent.CountDownLatch` or some other abstraction might be a little clearer.